### DISCUSSION TITLE
Effective value of consumer_timeout is not smaller than 60 seconds

### DISCUSSION THREAD

**Discussion by simpadjo** [rabbitmq/rabbitmq-server#5795] [Other] [answered]
Hello. I'm fixing a bug in a Scala client library for RabbitMq. 
I want to write a test that verifies that my code correctly reacts when Rabbit server closes the channel due to not receiving an ack.
To make the test complete in reasonable time, I set `consumer_timeout = 2000` and expect that the channel will be closed after 2 seconds. But server logs an error only after 60 seconds.
I tried to change many parameters including `heartbeat` and `net_ticktime` but it didn't help.
Am I missing something?

**Comment by lukebakken** [comment, marked as answer]
By default, RabbitMQ only checks every 60 seconds to see if consumers have exceeded the timeout. This is configured by the `channel_tick_interval` setting.

So, if for your test you'd like to have 2000ms consumer timeout, you need to also adjust `channel_tick_interval` via the `advanced.config` file:

```
[
    {rabbit, [
        {channel_tick_interval, 1000}
    ]}
].
```

If you're curious where these values are used, see the following:

https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbit/src/rabbit_channel.erl#L2793-L2795

https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbit/src/rabbit_channel.erl#L848-L858

https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbit/src/rabbit_channel.erl#L2827-L2849

**Comment by lukebakken** [comment]
If you don't mind, link to the bug you're fixing as well as the work you're doing. Thanks!

**Comment by simpadjo** [comment]
@lukebakken thank you! I will try out your advice later.
Here is the issue I faced in prod recently https://github.com/profunktor/fs2-rabbit/issues/736

**Comment by simpadjo** [comment]
@lukebakken I solved my issue, thank you once again

**Reply by lukebakken** [reply]
Thank you for letting us know!
